scheduler/services/resource/ocs_resource_service.py

```diff
@@ -249,6 +259,18 @@ def _load_csv(self, site: Site, name: str, c: Callable[[List[str]], Set[str]]) -
                 # Advance the previous row date where data was defined.
                 prev_row_date = row_date
 
+        if file:
+            data_str = file.decode()
+            f = StringIo(data_str)
+            _process_file(f)
+        elif name:
+             with open(os.path.join(self._path, name)) as f:
+                 _process_file(f)
+        else:
+            raise ValueError('Both name and file are None')
```

Comment:
Not sure what's going on here.

Reply:
`csv` does not accept BytesIO only StringIO.

Reply:
Okay... makes sense. Instead of passing in both `name: Optional[str]` and `file: Optional[BytesIO]`, we should probably pass in one variable:
```python
input: Union[str, BytesIO]
```
or, in modern Python fashion:
```python
input: str | BytesIO
```
and then process it based on type, maybe using `match`. This is another case where we have four possible combinations of values (two for each `Optional`, where it makes no sense if both or neither are passed in) when we should only have two.